clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -24,123 +24,380 @@ package common
 )
 
 func init() {
 	measurement.Register(waitForControlledPodsRunningName, createWaitForControlledPodsRunningMeasurement)
 }
 
 func createWaitForControlledPodsRunningMeasurement() measurement.Measurement {
-	return &waitForControlledPodsRunningMeasurement{}
+	return &waitForControlledPodsRunningMeasurement{
+		queue:      workqueue.New(),
+		checkerMap: make(map[string]*objectChecker),
+	}
 }
 
-type waitForControlledPodsRunningMeasurement struct{}
+type waitForControlledPodsRunningMeasurement struct {
+	informer          cache.SharedInformer
+	kind              string
+	namespace         string
+	labelSelector     string
+	fieldSelector     string
+	operationTimeout  time.Duration
+	stopCh            chan struct{}
+	isRunning         bool
+	queue             workqueue.Interface
+	workerGroup       wait.Group
+	handlingGroup     wait.Group
+	lock              sync.Mutex
+	opResourceVersion uint64
+	checkerMap        map[string]*objectChecker
+}
 
-// Execute waits until all specified RCs have all pods running or until timeout happens.
-// RCs can be specified by field and/or label selectors.
+// Execute waits until all specified controlling objects have all pods running or until timeout happens.
+// Controlling objects can be specified by field and/or label selectors.
 // If namespace is not passed by parameter, all-namespace scope is assumed.
-func (*waitForControlledPodsRunningMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
+// "Start" action starts observation of the controlling objects, while "gather" waits for until
+// specified number of controlling objects have all pods running.
+func (w *waitForControlledPodsRunningMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
 	var summaries []measurement.Summary
-	kind, err := util.GetString(config.Params, "kind")
+	action, err := util.GetString(config.Params, "action")
 	if err != nil {
 		return summaries, err
 	}
-	namespace, err := util.GetStringOrDefault(config.Params, "namespace", metav1.NamespaceAll)
+
+	switch action {
+	case "start":
+		w.kind, err = util.GetString(config.Params, "kind")
+		if err != nil {
+			return summaries, err
+		}
+		w.namespace, err = util.GetStringOrDefault(config.Params, "namespace", metav1.NamespaceAll)
+		if err != nil {
+			return summaries, err
+		}
+		w.labelSelector, err = util.GetStringOrDefault(config.Params, "labelSelector", "")
+		if err != nil {
+			return summaries, err
+		}
+		w.fieldSelector, err = util.GetStringOrDefault(config.Params, "fieldSelector", "")
+		if err != nil {
+			return summaries, err
+		}
+		w.operationTimeout, err = util.GetDurationOrDefault(config.Params, "operationTimeout", defaultOperationTimeout)
+		if err != nil {
+			return summaries, err
+		}
+		return summaries, w.start(config.ClientSet)
+	case "gather":
+		syncTimeout, err := util.GetDurationOrDefault(config.Params, "syncTimeout", defaultWaitForRCPodsTimeout)
```

Comment:
defaultWaitForRCPodsTimeout name is misleading

defaultSyncTimeout maybe?

[BTW - I like the syncTimeout name]

Reply:
Done.